rules/regexp-capture.yml

```diff
@@ -0,0 +1,31 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/ast-grep/ast-grep/main/schemas/rule.json
+
+id: regexp-capture
+message: Use '&nosub' regexps if not using capture groups
+severity: error # error, warning, info, hint
+language: spicy
+rule:
+  kind: "regexp"
+  pattern: "$RE"
+  # Assume every occurrance of a pair of opening/closing parens corresponds
+  # to a capture. This e.g., ignores the existance of escaped parens.
+  regex: "/\\(.*\\)/"
+  all:
+    - inside:
+        kind: "field_decl"
+    - not:
+        precedes:
+          kind: "attribute"
+          regex: "&nosub"
+    - not:
+        precedes:
+          kind: "block"
```

Comment:
Obvious issue here: if the field hook is not defined inline we currently do not see it.

Reply:
I expanded the note to document this false positive and show a workaround.